File: Sqrt(x)/Sqrt(x).py


class Solution:
    # @param x, an integer
    # @return an integer
    def sqrt(self, x):
        if x==0:
            return 0
        pre=0
        cur = 1
        while abs(cur-pre)>1e-5:
            pre=cur
            cur=pre/2.0 + x/(2.0*pre)
        return int(cur)
 

File: solutions/Sqrt(x)/Sqrt(x).cpp

class Solution {
public:
    int sqrt(int x) {
        if(x==0)
            return 0;
        double pre;
        double cur=1;
        while(abs(cur-pre)>1e-5)
        {
            pre=cur;
            cur=pre/2 + x/(2*pre);
        }
        return int(cur);
    }
};


 